storagemarket/provider_loop.go

```diff
@@ -42,6 +43,41 @@ func (p *Provider) loop() {
 			deal := dealReq.deal
 			log.Infow("process accept deal request", "id", deal.DealUuid)
 
+			// get current sealing pipeline status
```

Comment:
Refactoring suggestion:
In several places in this case block we have code that
- checks for an error
- creates ProviderDealRejectionInfo with the error
- sends an acceptDealResp on the response channel
- `continue`s

It may be easier to follow the logic if we can make it look more like the traditional go form of `if err != nil { return err }`. So for example if we put all the logic in this case block into a function something like `processDealRequest(dealReq) error` then we can do
```
	err := processDealRequest(dealReq)
	if err != nil {
		ri := &api.ProviderDealRejectionInfo{Accepted: false, Reason: err.Error()}
		dealReq.rsp <- acceptDealResp{ri: ri, err: nil}
		continue
	}
```

Reply:
Good idea, but we actually need both `err` and `reason`. Have a look at the refactor and let me know if this is what you had in mind.